[docs] Update CodePush migration channel surfing verbiage (#45255)
expo/expo: docs/pages/eas-update/codepush.mdx

**CodePush has single streams of updates for deployments**. What this means is that you can point a build to a deployment, and it will pull updates from that. If you want to change the deployment that is targeted by a build, you can do this at runtime through a JavaScript API.

> **EAS Update has multiple streams of updates** &mdash; one that correspond to your source control branches (called branches), and another called channels, which point to branches. The mapping between channels and branches is handled on the server side, and a channel can point to different branches for each runtime version (additionally, more advanced logic may be expressed, such as to support incremental rollouts). Builds are not directly associated with branches, but rather with channels. Each build points to a single channel by default. The reason for this is that it ensures that certain branches (for example: development, staging) don't automatically go out to production, so your preview updates don't go to your production users. This helps you separate the two main uses of EAS Update: previews and production hotfixes.

Reviewer: ```suggestion
**EAS Update has multiple streams of updates** &mdash; one that corresponds to your source control branches (called branches), and another called channels, which point to branches. The mapping between channels and branches is handled on the server side, and a channel can point to different branches for each runtime version (additionally, more advanced logic may be expressed, such as to support incremental rollouts). Builds are not directly associated with branches, but rather with channels. Each build points to a single channel by default. The reason for this is that it ensures that certain branches (for example: development, staging) don't automatically go out to production, so your preview updates don't go to your production users. This helps you separate the two main uses of EAS Update: previews and production hotfixes.
```

</Collapsible>

<Collapsible summary="Differences in how updates are selected at runtime">

A key distinction between CodePush and EAS Update that can impact your release process is that **with CodePush, the client controls the target update deployment at runtime**, and **with EAS Update, the default path is controlled on the server side, by mapping channels to branches**. In the default EAS Update flow, a build requests updates for its configured channel, and EAS Update returns the latest compatible update from the branch mapped to that channel.

> The ability to control the target deployment at runtime is commonly used with CodePush in staging environments to allow non-technical stakeholders to test features from a single build on Google Play Beta and TestFlight. With EAS Update, you can use [channel surfing](/eas-update/override/#what-is-channel-surfing) to switch update channels in production builds at runtime. Development builds can also load updates from any compatible channel and are a good fit for developer-focused testing.

Reviewer: ```suggestion
The ability to control the target deployment at runtime is commonly used with CodePush in staging environments to allow non-technical stakeholders to test features from a single build on Google Play Beta and TestFlight. With EAS Update, you can achieve the same via [channel surfing](/eas-update/override/#what-is-channel-surfing). This allows you to switch update channels in non-development builds at runtime. Development builds can also load updates from any compatible channel and are a good fit for developer-focused testing.
```

Suggesting "non-development builds" as an alternative to "production builds" because we also have internal distribution / preview  builds which also support channel surfing.
